### ISSUE TITLE
Not able to change Timeout configuration and values.

### ISSUE THREAD

**Issue by saurabh-tiket** [envoyproxy/gateway#1509]
We are evaluating Envoy as our API gateway, During our POC we realized one of our backends was taking some time to produce a response (more than 15 seconds) and the envoy proxy giving an `upstream timeout` error. 

While trying to change this value during manifests, I realized it is not possible. I checked the `Policy Attachment` [documentation](https://github.com/envoyproxy/gateway/blob/1907687b3c76d72073310e54333226bcb3bee96e/docs/latest/design/envoy-patch-policy.md) and realised there is no way to change this value.  this is the value I need to change 

https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-timeout

Can someone confirm whether it is possible to change this value? I also see a similar issue here https://github.com/envoyproxy/gateway/issues/1169 

I tried reading the SPEC of `EnvoyPatchPolicy` CRD and I think it is not supported 

```
                      description: Type is the typed URL of the Envoy xDS Resource
                      enum:
                      - type.googleapis.com/envoy.config.listener.v3.Listener
                      - type.googleapis.com/envoy.config.route.v3.RouteConfiguration
                      - type.googleapis.com/envoy.config.cluster.v3.Cluster
                      - type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment
                      type: string
```
I tested with the following manifest to apply `EnvoyPatchPolicy`, (this is not the correct manifest and needs an update) 

```
apiVersion:  gateway.envoyproxy.io/v1alpha1
kind: EnvoyPatchPolicy
metadata:
  name: eg-timeout-policy
  namespace: infra-ns
status: "True"
spec:
  priority: 0
  jsonPatches:
  - operation: 
      op: add
      path: timeout
      value: 10s
    name: Route
    type: type.googleapis.com/envoy.config.route.v3.RouteConfiguration
  targetRef:
    group: ""
    kind: Gateway
    name: eg
  type: JSONPatch
```

--- 

Also, I tried applying a test `EnvoyPatchPolicy` and wasn't able to succeed.  I think the documentation on how to use `EnvoyPatchPolicy` is not clear.  

The example I used is https://github.com/envoyproxy/gateway/blob/1907687b3c76d72073310e54333226bcb3bee96e/docs/latest/design/envoy-patch-policy.md 

Following is the error I'm getting 

```
 $ kubectl apply -f envoy-test-policy.yaml 
The EnvoyPatchPolicy "ratelimit-patch-policy" is invalid: 
* spec.priority: Required value
* status: Required value
* 
```

---

Can someone help us to confirm whether it is possible to change [this timeout ](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-timeout)value ?? It is a blocker for us to use envoy in production.

**Comment by arkodg** [comment]
hey @saurabh-tiket EnvoyPatchPolicy is still WIP and will be part of v0.5.0.
Lets keep this issue open to ensure that the core issue is resolved - you can configure a route timeout to the backend.
Also relates to https://github.com/envoyproxy/gateway/issues/1492

**Comment by saurabh-tiket** [comment]
Hey @arkodg 

Thanks for the quick reply! It's great to know that EnvoyPatchPolicy is still cooking and will be served in v0.5.0. Definitely looking forward to being able to tinker with the route timeout to the backend.

Issue #1492 does look related, I'll be keeping an eye on both these threads.

By the way, do you have any rough timeline on when v0.5.0 might drop? Just trying to plan things out on my end.

Cheers

**Comment by arkodg** [comment]
@saurabh-tiket [this](https://github.com/envoyproxy/gateway/milestone/13) milestone link should have more details.

**Comment by saurabh-tiket** [comment]
@arkodg Ok thanks

**Comment by arkodg** [comment]
blocked on https://gateway-api.sigs.k8s.io/geps/gep-1742/ moving into `Experimental`

**Comment by arkodg** [comment]
closing in favor of https://github.com/envoyproxy/gateway/issues/877
